datafusion/sql/src/planner.rs

```diff
@@ -2184,7 +2178,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
     fn sql_interval_to_literal(
         &self,
-        value: String,
+        value: SQLExpr,
```

Comment:
Intervals now support arbitrary expressions, not just strings: https://github.com/sqlparser-rs/sqlparser-rs/pull/517